### DISCUSSION TITLE
Homing X + Y axis only

### DISCUSSION THREAD

**Discussion by brad1983** [winder/Universal-G-Code-Sender#2116] [Help] [answered]
Is this possibly as I only have two limit switches?

**Comment by breiler** [comment, marked as answer]
Depends on your controller. UGS will only issue a homing command and it is up to the controller how to handle it.

For GRBL, read the configuration section about homing here:
https://github.com/gnea/grbl/blob/master/grbl/config.h#L90

**Reply by brad1983** [reply]
Great start thanks.
I will assume I can find this file within the UGS files?

**Reply by breiler** [reply]
No, the link that I posted was the config for the GRBL firmware. UGS can generally not do much about altering the homing process/sequence. 

Some controller firmwares does allow homing each axis separatly. But as you did not provide what controller you have I can not help you much more than this.

**Reply by brad1983** [reply]
Is this the info you require? It's for a yorahome 3018 pro.

Master chip: ATMEGA 328P (Arduino Nano)

**Comment by chrisla23** [comment]
I would assume you are on grbl if it is Arduino based. If you open the
console winder in UGS while connected and idle and hit enter a few times it
will give a clue.

If it is grbl you would need to modify build arguments in the config file,
compile, and flash a custom grbl build onto the controller. You would then
have some non-standard commands available to home one axis at a time. All
of this is somewhat intermediate in skill level. It's not huge to someone
who knows the ins and out of Arduino programming and grbl but it has a
number of dependencies. You would also want to be aware of the grbl version
(and flash the same one) and have a backup of the various tuneable
settings.

-Chris
